RiotSharp/Endpoints/ClientEndpoint/ClientEndpoint.cs

```diff
@@ -0,0 +1,170 @@
+                                       .ConfigureAwait(false);
+            return JsonConvert.DeserializeObject<PlayerMainRunes>(json);
+        }
+
+        /// <inheritdoc />
+        public async Task<PlayerSummonerSpellList> GetPlayerSummonerSpellsAsync(string summonerName)
+        {
+            var json = await _requester.CreateGetRequestAsync(Host, string.Format($"{ClientDataRootUrl}{PlayerSummonerSpellsBySummonerNameUrl}", summonerName))
+                                       .ConfigureAwait(false);
+            return JsonConvert.DeserializeObject<PlayerSummonerSpellList>(json);
+        }
+
+        /// <inheritdoc />
+        public async Task<PlayerScores> GetPlayerScoresAsync(string summonerName)
+        {
+            var json = await _requester.CreateGetRequestAsync(Host, string.Format($"{ClientDataRootUrl}{PlayerScoresBySummonerNameUrl}", summonerName))
+                                       .ConfigureAwait(false);
+            return JsonConvert.DeserializeObject<PlayerScores>(json);
+        }
+
+        /// <inheritdoc />
+        public async Task<GameEventList> GetGameEventListAsync()
+        {
+            var json = await _requester.CreateGetRequestAsync(Host, $"{ClientDataRootUrl}{GameEventListUrl}").ConfigureAwait(false);
+            return JsonConvert.DeserializeObject<GameEventList>(json);
+        }
+
+        /// <inheritdoc />
+        public async Task<GameStats> GetGameStatsAsync()
+        {
+            var json = await _requester.CreateGetRequestAsync(Host, $"{ClientDataRootUrl}{GameStatsUrl}").ConfigureAwait(false);
+            return JsonConvert.DeserializeObject<GameStats>(json);
+        }
+
+        /// <summary>
+        ///     Gets the singleton instance of the <see cref="ClientEndpoint" /> class.
+        /// </summary>
+        /// <returns>The singleton instance of the <see cref="ClientEndpoint" /> class.</returns>
+        public static IClientEndpoint GetInstance()
+        {
+            if (Requesters.ClientApiRequester == null)
+            {
+                var clientHandler = new HttpClientHandler
+                                    {
+                                        ServerCertificateCustomValidationCallback = ValidateClientCertificate
+                                    };
+                Requesters.ClientApiRequester = new Requester(clientHandler);
+            }
+
+            return _instance ?? (_instance = new ClientEndpoint(Requesters.ClientApiRequester));
+        }
+
+        private static bool ValidateClientCertificate(HttpRequestMessage message, X509Certificate2 certificate, X509Chain chain, SslPolicyErrors errors)
+        {
+            if (errors == SslPolicyErrors.None)
+            {
+                return true;
+            }
+
+            if (certificate?.Thumbprint?.Equals(ClientCertificateThumbprint, StringComparison.OrdinalIgnoreCase) == true)
```

Comment:
```suggestion
            if (certificate?.Thumbprint?.Equals(ClientCertificateThumbprint, StringComparison.OrdinalIgnoreCase))
```

Reply:
I'm using a null-propagation check here, so it evaluates into a `bool?` instead of a plain `bool` and, thus, I cannot fall back to your suggestions.